### PR TITLE
Added dropdowns and interactive gallery to see participants

### PR DESCRIPTION
**Pull request by Kenneth-Funes-TU-Delft**
- The proposed AnnotationTypeDropdown did not add any new functionality that the base type Dropdown did not have, so it was removed.
- Instantiated Dropdown now for both action selection and participant selection, while managing their state in Redux
- Added custom styling and trying to have a functional sidebar with the video playback on the right
- Added interactive gallery as a fullscreen overlay
- The gallery image is the image reference for annotators to identify participants.
- A button was added to display the gallery
- The gallery is displayed as a modal overlay (fullscreen, steal keyboard input)
- Added a simple logic to set the participant to annotate based on a click on the gallery image. This is easy given the gallery image is a regular grid.
- Note: the image was made svg because there seem to be rules in the project in which only svg images are picked up. Unclear how to change atm

### CODE REVIEW

**Comment by Kenneth-Funes-TU-Delft** [comment]
Here is the state of the ui for action annotations. The next step is to capture video playback frames. It does not seem obvious to get a callback for each specific frame and I'm trying to reuse functionality from the old keypoints task. However, what seems to have been generating callbacks for each frame is tighly linked to OpenCV with OpencvFlowPlayer , which I believe we want to avoid at this stage.

File: covfee/client/tasks/action_annotation/index.tsx [hunk 1]
```diff
@@ -7,14 +7,23 @@ import { AllPropsRequired } from "../../types/utils";
...
-import { MenuProps } from "antd";
+import type { MenuProps } from "antd";
+import { Dropdown, Menu, Space, MenuInfo, Button } from "antd";
+import { DownOutlined } from "@ant-design/icons";
 import styles from "./action_annotation.module.css";
 
+// You can download this image and place it in the art folder from:
+// @helix.ewi.tudelft.nl:/home/kfunesmora/mingle/media
+// Do not commit to the repo for confidentiality reasons.
+import ConflabGallery from "../../art/conflab-gallery.svg";
```
Comment (Era-Dorta-TU-Delft) [inline covfee/client/tasks/action_annotation/index.tsx:18]: Not sure if we want to have the path relative to the repo. Also I think the default for web stuff is to put those files under `media`

File: covfee/client/tasks/action_annotation/index.tsx [hunk 2]
```diff
@@ -7,14 +7,23 @@ import { AllPropsRequired } from "../../types/utils";
...
+
 interface Props extends CovfeeTaskProps<ActionAnnotationTaskSpec> {}
 
 const annotation_types: string[] = ["speaking", "laughing"];
-
+const participants_available: string[] = [];
+for (let i = 1; i < 51; i++) {
```
Comment (Era-Dorta-TU-Delft) [inline covfee/client/tasks/action_annotation/index.tsx:24]: Will the 51 be reused? It might be good to put it in a constant.

**Review by Era-Dorta-TU-Delft — APPROVED** [review]
I didn't look in a lot of detail but it seems fine. Regarding the frame number, on the videojs class I see a `currentTime()` and `playbackRate()` methods. Can we query those to go from times to frame numbers?

**Comment by Era-Dorta-TU-Delft** [comment]
For future PRs, please select me as a reviewer on the top right toggle. Otherwise, I don't get any email notifications.

File: covfee/client/tasks/action_annotation/index.tsx (hunk 1 above)
Comment (Kenneth-Funes-TU-Delft) [reply covfee/client/tasks/action_annotation/index.tsx:18]: I agree. However, I tried to be consistent with the way covfee is currently designed and distributed. Example: https://github.com/Era-Dorta-TU-Delft/covfee/blob/master/covfee/client/admin/login.tsx#L7

**Comment by Kenneth-Funes-TU-Delft** [comment]
Indeed, there are `currentTime()` and `playbackRate()` methods. However, I prefer to end up with a solution in which the video player is the one driving the callbacks, instead of constantly querying for the current state. That seems to be the challenge. I've been studying the implementation of `OpencvFlowPlayer` in the repo which provides some hints. The way it was done was by attaching OpenCV to the the videojs instance and then you end up with a callback per frame.

File: covfee/client/tasks/action_annotation/index.tsx (hunk 2 above)
Comment (Kenneth-Funes-TU-Delft) [reply covfee/client/tasks/action_annotation/index.tsx:24]: In reality the "51" is not what matters. The actual constant is the `participants_available`,as a list of strings, which could have been ["PersonA", "PersonB", ...].  That being said, that should be noted by making their names uppercase. I'll do that, including annotation_types.

**Pull request merged**